### PR TITLE
feat: Add measurement map selector with geometry selection

### PR DESCRIPTION
**Pull request by benjaminhuth**
Allows to create a subset of a particles-measurement-map with a geometry selection.

### CODE REVIEW

**Review by andiwand — DISMISSED** [review]

**Comment by andiwand** [comment]
I wonder if such a algorithm is easy enough to be implemented in python. but not sure how efficient that would be and if we have all the necessary bindings. just a thought

File: Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  /// @param cfg is the config struct to configure the algorithm
+  /// @param level is the logging level
+  MeasurementMapSelector(Config cfg, Acts::Logging::Level lvl)
+      : IAlgorithm("MeasurementMapSelector", lvl), m_cfg(cfg) {
+    m_inputSourceLinks.initialize(m_cfg.inputSourceLinks);
+    m_inputMap.initialize(m_cfg.inputMeasurementParticleMap);
+    m_outputMap.initialize(m_cfg.outputMeasurementParticleMap);
+  }
+
+  virtual ~MeasurementMapSelector() {}
```
Comment (paulgessinger) [inline Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp:57]:
```suggestion
  virtual ~MeasurementMapSelector() = default;
```

**Review by paulgessinger — APPROVED** [review]
Ok

**Comment by benjaminhuth** [comment]
> I wonder if such a algorithm is easy enough to be implemented in python. but not sure how efficient that would be and if we have all the necessary bindings. just a thought

I wonder if it is really worth to deal with all these things like binding the whiteboard and the EDM etc...
But in principle it would be nice to have some more flexibility on manipulating the EDM in python...

File: Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp (hunk 1 above)
Comment (benjaminhuth) [reply Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp:57]: Hmm I agree that would be cleaner, but if the CI runs through I would keep it as it is I guess

Comment (andiwand) [reply Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp:57]: clang tidy may be against it :smile:

Comment (benjaminhuth) [reply Examples/Algorithms/Utilities/include/ActsExamples/Utilities/MeasurementMapSelector.hpp]: Okay, but then I need a new review :laughing:

**Review by andiwand — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6381163)!

|status|job|
|------|---|
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33390055) |
|:red_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389154) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389153) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389152) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389151) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389150) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389149) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389148) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389147) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389146) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389145) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389144) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389143) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389142) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389140) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33389139) |

**Pull request merged**